src/uct/tcp/tcp_cm.c

```diff
@@ -160,20 +161,30 @@ ucs_status_t uct_tcp_cm_send_event(uct_tcp_ep_t *ep, uct_tcp_cm_conn_event_t eve
                 (ep->conn_state != UCT_TCP_EP_CONN_STATE_CONNECTED),
                 "ep=%p", ep);
 
-    pkt_length        = sizeof(*pkt_hdr);
+    pkt_length                  = sizeof(*pkt_hdr);
     if (event == UCT_TCP_CM_CONN_REQ) {
-        cm_pkt_length = sizeof(*conn_pkt);
+        cm_pkt_length           = sizeof(*conn_pkt);
+
+        if (ep->conn_state == UCT_TCP_EP_CONN_STATE_CONNECTING) {
```

Comment:
is it possible to send  `UCT_TCP_CM_CONN_REQ` in non `UCT_TCP_EP_CONN_STATE_CONNECTING` state?

Reply:
yes, when UCT/TCP EP was in CONNECTED state after that EP was created from accepting the connection and then the user wants to create new EP to the same peer (we found this EP from khash and send CONN_REQ mesage to the peer to enable RX capability on peer's side)

Reply:
maybe always sending magic number would be simpler?